src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+import { SetTodoModalType } from './types/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoModal, setTodoModal] = useState<SetTodoModalType>({
+    todo: {
+      completed: false,
+      id: 0,
+      title: '',
+      userId: 0,
+    },
+    user: {
+      email: '',
+      id: 0,
+      name: '',
+      phone: '',
+    },
+  });
+  const [isClicked, setIsClicked] = useState(false);
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterMode, setFilterMode] = useState<string>('All');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const extractTodos = async () => {
+    try {
```

Comment:
Consider renaming 'extractTodos' to a more descriptive name, such as 'fetchTodos'.